autoconfigure/collector-eventhub/src/test/java/zipkin/autoconfigure/collector/eventhub/EventHubCollectorAutoConfigurationTest.java

```diff
@@ -31,184 +31,181 @@
-
-        String storageBlobPrefix = "pashmak";
-
-        context = new AnnotationConfigApplicationContext();
-        addEnvironment(context, "zipkin.collector.eventhub.storageConnectionString:" + dummyEventHubConnectionString);
-        addEnvironment(context, "zipkin.collector.eventhub.eventHubConnectionString:" + dummyStorageConnectionString );
-        addEnvironment(context, "zipkin.collector.eventhub.storageBlobPrefix:" + storageBlobPrefix);
-        context.register(PropertyPlaceholderAutoConfiguration.class,
-                EventHubCollectorAutoConfiguration.class,
-                InMemoryConfiguration.class);
-        context.refresh();
-
-        EventHubCollector collector = context.getBean(EventHubCollector.class);
-        assertNotNull(collector);
-        assertEquals(storageBlobPrefix, collector.getStorageBlobPrefix());
-    }
-
-    @Test
-    public void provideCollectorComponent_canSetStorageContainerName() {
-
-        String storageContainerName = "pashmak";
-
-        context = new AnnotationConfigApplicationContext();
-        addEnvironment(context, "zipkin.collector.eventhub.storageConnectionString:" + dummyEventHubConnectionString);
-        addEnvironment(context, "zipkin.collector.eventhub.eventHubConnectionString:" + dummyStorageConnectionString );
-        addEnvironment(context, "zipkin.collector.eventhub.storageContainerName:" + storageContainerName);
-        context.register(PropertyPlaceholderAutoConfiguration.class,
-                EventHubCollectorAutoConfiguration.class,
-                InMemoryConfiguration.class);
-        context.refresh();
-
-        EventHubCollector collector = context.getBean(EventHubCollector.class);
-        assertNotNull(collector);
-        assertEquals(storageContainerName, collector.getStorageContainerName());
+  AnnotationConfigApplicationContext context;
+
+  static final String dummyEventHubConnectionString = "endpoint=sb://someurl.net;SharedAccessKeyName=dumbo;SharedAccessKey=uius7y8ewychsih";
+  static final String dummyStorageConnectionString = "a=b;c=d";
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @After
+  public void close() {
+    if (context != null) context.close();
+  }
+
+  @Test
+  public void doesntProvideCollectorComponent_whenEventHubConnectionStringUnset() {
+    context = new AnnotationConfigApplicationContext();
+    context.register(PropertyPlaceholderAutoConfiguration.class,
+        EventHubCollectorProperties.class,
+        EventHubCollectorAutoConfiguration.class,
+        InMemoryConfiguration.class);
+    context.refresh();
+
+    thrown.expect(NoSuchBeanDefinitionException.class);
+    context.getBean(EventHubCollector.class);
+  }
+
+  @Test
+  public void provideCollectorComponent_whenEventHubConnectionStringIsSet() {
+    context = new AnnotationConfigApplicationContext();
+    addEnvironment(context, "zipkin.collector.eventhub.storageConnectionString:" + dummyStorageConnectionString);
```

Comment:
Most likely, a YAML file will be used so the verbosity will be reduced. But sure can change. I personally prefer descriptive - bear in mind for example there are two eventHub properties.